src/mount.ts

```diff
@@ -133,6 +137,46 @@ export function mount(originalComponent: any, options?: MountingOptions) {
   const { emitMixin, events } = createEmitMixin()
   vm.mixin(emitMixin)
 
+  // stubs
+  if (options?.global?.stubs) {
+    transformVNodeArgs((args) => {
+      // regular HTML Element. Do not stubs these
+      if (Array.isArray(args) && typeof args[0] === 'string') {
```

Comment:
Do you think we could extract all this assertions+comments into inline functions? I think it would make each early return cleaner:

```js
      if (isHtmlElement(args) || isComment(args) || isParentComponent(args)) {
````

or similar ^^

same thing with the if statement in lines 164 and 168, too.


Just my two cents 😇 feel free to resolve this if you think this will either be simple enough to understand, or change significantly so this refactor does not make sense.

Reply:
yep. sounds good. I was so focused on getting it to work I lost track of how unreadable it was getting. let's keep this codebase clean and understandable!

Reply:
(if anyone is reading this who would like to make this refactor before I do feel free to make a PR targeting this one)

Reply:
@afontcu this has been done lmk if you have any feedback.

I have no problem making further changes if it improves the quality of the codebase

Reply:
I think this is completely fine for now! 😄 Thanks for this. Once we've established a good foundations in terms of features, we can try and see the whole codebase using a bird's-eye view and see where we can improve readalibity and maintainability.